src/extensions/common/useFeedReducer.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useMemo, useReducer } from "react"
+
+export interface AppendOnlyLog {
+    appendMessage: (msg: any) => void
+    allMessages: () => any[]
+    onMessage: (callback: (msg: any) => void) => void
+}
+
+export const useFeedReducer = <State, Action>(reducer: (s: State, a: Action) => State, initialState: State, subfeed: AppendOnlyLog | null): [State, (a: Action) => void] => {
```

Comment:
A wrapper hook to a reducer that redirect actions to a subfeed on the server - when updates to the subfeed come back to the client, they are played out on `reducer`